src/credentials/awsCredentialsStatusBarItem.ts

```diff
@@ -33,10 +39,26 @@ export async function initializeAwsCredentialsStatusBarItem(
     )
 }
 
-export function updateCredentialsStatusBarItem(statusBarItem: vscode.StatusBarItem, credentialsId?: string) {
-    statusBarItem.text = localize(
-        'AWS.credentials.statusbar.text',
-        'AWS: {0}',
-        credentialsId ?? STATUSBAR_TEXT_NO_CREDENTIALS
+// Resolves when the status bar reaches its final state
+export async function updateCredentialsStatusBarItem(statusBarItem: vscode.StatusBarItem, credentialsId?: string) {
+    clearTimeout(timeoutID)
+
+    // Shows confirmation text in the status bar message
+    let delay = 0
+    if (credentialsId) {
+        delay = STATUSBAR_CONNECTED_DELAY
+        statusBarItem.text = localize('AWS.credentials.statusbar.text', 'AWS: {0}', STATUSBAR_TEXT_CONNECTED)
+    }
+
+    return new Promise<void>(
+        resolve =>
+            (timeoutID = setTimeout(() => {
+                statusBarItem.text = localize(
+                    'AWS.credentials.statusbar.text',
+                    'AWS: {0}',
+                    credentialsId ?? STATUSBAR_TEXT_NO_CREDENTIALS
```

Comment:
If `credentialsId` is defined, you'll first write "AWS: (connected)" followed by "AWS: <credentialsId>". Is this expected?

Reply:
Yes. Adding some extra movement to the status bar will help bring attention to successful connections while not being as intrusive as a notification pop-up.